chore(curriculum): add css animations transcripts (#58375)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-animations-and-accessibility/672cf764cf55a70433590def.md

Additionally, animations can be distracting for users with cognitive disabilities or attention disorders. Rapid flashing or strobing effects are particularly problematic. They can trigger seizures in people with photosensitive epilepsy. As a general rule, avoid any content that flashes more than three times per second.

Another issue is that animations can make it difficult for some users to focus on or read content. This is especially true for users with low vision or reading difficulties who may struggle to track moving text or shifting layouts.

To address these concerns, CSS provides the `prefers-reduced-motion` media query. This feature allows web developers to detect if the user has requested minimal animations or motion effects at the system level.

Here's how you can use `prefers-reduced-motion`:

```css
@media (prefers-reduced-motion: reduce) {
  * {
    animation-duration: 0.01ms !important;
    animation-iteration-count: 1 !important;
    transition-duration: 0.01ms !important;
    scroll-behavior: auto !important;
  }
}
```

> This CSS code snippet effectively disables most animations and transitions for users who have indicated a preference for reduced motion. Let's break it down:
> 
> The `@media` query rule checks if the user prefers reduced motion. If true, it applies the enclosed styles.
> 
> Inside the media query, we're targeting all elements (`*`) and overriding animation and transition properties.
> 
> We set `animation-duration` and `transition-duration` to an extremely small value (`0.01ms`). This essentially turns off the animations while still allowing them to complete, which can be important for certain functionality.

Reviewer: ```suggestion
This CSS code snippet effectively disables most animations and transitions for users who have indicated a preference for reduced motion. Broken down:

The `@media` query rule checks if the user prefers reduced motion. If true, it applies the enclosed styles.

Inside the media query, all elements (`*`) are targeted and their animation and transition properties overridden.

The `animation-duration` and `transition-duration` properties are set to a relatively small value (`0.01ms`). This essentially turns off the animations while still allowing them to complete, which can be important for certain functionality.
```

`animation-iteration-count: 1` ensures that any looping animations only play once.

`scroll-behavior: auto` disables smooth scrolling effects.

The `!important` declaration is used to ensure these rules take precedence over other animation styles.

It's important to note that while this method effectively reduces motion, it's a blanket approach. For more precise control, you might want to define specific reduced-motion alternatives for your animations.

Here's an example of a more targeted approach:

```css
.animated-element {
  transition: transform 0.3s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .animated-element {
    transition: none;
  }
}
```

> In this case, we're only disabling the `transition` for a specific element when reduced motion is preferred. This allows you to provide alternative, less motion-intensive experiences for users who need them.

Reviewer: ```suggestion
In this case, only the `transition` is disabled for a specific element when reduced motion is preferred. This allows you to provide alternative, less motion-intensive experiences for users who need them.
```
